cmds/cpu/cpu9p_unix.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !windows && !plan9
+// +build !windows,!plan9
+
+package main
+
+import (
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+// lock implements p9.File.Lock.
+func (l *cpu9p) Lock(pid, locktype, flags int, start, length uint64, client string) error {
+	var err error
+	// shit. Not approved yet.
+	// switch p9.LockType(locktype) {
+	// case p9.ReadLock, p9.WriteLock:
+	// 	if err := unix.Flock(int(l.file.Fd()), unix.LOCK_EX); err != nil {
+	// 		err = p9.ELockError
+	// 	}
+	// case p9.Unlock:
+	// 	if err := unix.Flock(int(l.file.Fd()), unix.LOCK_EX); err != nil {
+	// 		err = p9.ELockError
+	// 	}
+	// default:
+	// 	// 9P2000.L not only does not return an Rerror per standard,
+	// 	// it does not have a way to say "not implemented" for a Lock type.
+	// 	// They had 255 possible values, and use 3, ... ay yi yi.
+	// 	// 9P2000.L doesn't really fit the spirit of 9P.
+	// 	err = p9.ELockError
+	// }
+	switch locktype {
+	case 0, 1:
+		if err := unix.Flock(int(l.file.Fd()), unix.LOCK_EX); err != nil {
+			err = syscall.Errno(2)
+		}
+	case 2:
```

Comment:
what are the locktype values?  0-> read, 1->write, 2->unlock?  (based on the commented out stuff)

 it looks like you have the same code for case 2 as for 0 and 1.  if 2 == unlock, i'd have expected unix.Funlock or something.

Reply:
yeah that's a screwup on my part, will fix, good catch.